src/databricks/labs/remorph/reconcile/query_builder.py

```diff
@@ -0,0 +1,149 @@
+from io import StringIO
+
+from databricks.labs.remorph.reconcile.connectors.databricks import DatabricksDataSource
+from databricks.labs.remorph.reconcile.connectors.oracle import OracleDataSource
+from databricks.labs.remorph.reconcile.connectors.snowflake import SnowflakeDataSource
+from databricks.labs.remorph.reconcile.constants import (
+    ColumnTransformationType,
+    Constants,
+    SourceType,
+)
+from databricks.labs.remorph.reconcile.recon_config import (
+    ColumnMapping,
+    Schema,
+    Tables,
+    Transformation,
+    TransformRuleMapping,
+)
+
+
+class QueryBuilder:
+
+    def __init__(self, table_conf: Tables, schema: list[Schema], layer: str, source: str):
+        self.table_conf = table_conf
+        self.schema = schema
+        self.layer = layer
+        self.source = source
+
+    def build_hash_query(self) -> str:
```

Comment:
This function should support report-type `hash`

Reply:
made adaptible to report-type `hash`